Patch from Gaudenz Steinlin <gaudenz@debian.org>
Debian bug: #552014
Reported upstream: https://svn.boost.org/trac/boost/ticket/3544

The upstream bug was supposed to have been fixed in Changeset 59987
https://svn.boost.org/trac/boost/changeset/59987 on 2010-02-28 but
Boost 1.44 was released with part of the patch missing.  This 
abridged patch is that missing piece: add the tag to mpi python
libraries.

--- boost_1_41_0.orig/libs/mpi/build/Jamfile.v2
+++ boost_1_41_0/libs/mpi/build/Jamfile.v2
@@ -9,6 +9,7 @@
 #           Andrew Lumsdaine
 
 import mpi ;
+import indirect ;
 import python ;
 
 libraries = ;
@@ -18,8 +19,27 @@
 
 project boost/mpi
   : source-location ../src
+  : requirements
+    -<tag>@$(BOOST_JAMROOT_MODULE)%$(BOOST_JAMROOT_MODULE).tag
+    <tag>@$(__name__).tag
   ;
 
+rule tag ( name : type ? : property-set )
+{
+    local result = $(name) ;
+    if $(type) in STATIC_LIB SHARED_LIB IMPORT_LIB
+    {
+        if $(name) = boost_mpi_python && $(PYTHON_ID)
+        {
+            result = $(result)-$(PYTHON_ID) ;
+        }
+    }
+    
+    # forward to the boost tagging rule
+    return  [ indirect.call $(BOOST_JAMROOT_MODULE)%$(BOOST_JAMROOT_MODULE).tag 
+                $(result) : $(type) : $(property-set) ] ;
+}
+
 lib boost_mpi 
   : 
     broadcast.cpp
